File: corpus/java-efficiency-design-v1/changed/src/main/java/Enrichment.java
class ResultListFactory {
  static List<Item> create() { return new ArrayList<>(); }
}

class Enrichment {
  List<Item> enrich(List<Long> ids, Repository repository) {
    var results = ResultListFactory.create();
    for (var id : ids) {
      var item = repository.load(id);
      if (item.active()) {
        if (item.owner() != null) {
          if (item.owner().enabled()) results.add(item);
        }
      }
    }
    return results;
  }
}

File: corpus/dart-efficiency-design-v1/changed/lib/enrichment.dart
class ResultListFactory {
  static List<Item> create() => <Item>[];
}

Future<List<Item>> enrich(List<String> ids, Repository repository) async {
  final results = ResultListFactory.create();
  for (final id in ids) {
    final item = await repository.load(id);
    if (item.active) {
      if (item.owner != null) {
        if (item.owner!.enabled) results.add(item);
      }
    }
  }
  return results;
}
